packages/api/db/migration/20241016204145_create_animal_movement_task_tables.js

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+export const up = async (knex) => {
+  // Add new animal_movement entry to task_type table
+  await knex('task_type').insert({
+    task_name: 'Movement',
+    task_translation_key: 'MOVEMENT_TASK',
+  });
+
+  // Add location_id column to animal and animal_batch tables (foreign key to location table)
+  await knex.schema.alterTable('animal', (table) => {
+    table.uuid('location_id').references('location_id').inTable('location');
+  });
+  await knex.schema.alterTable('animal_batch', (table) => {
+    table.uuid('location_id').references('location_id').inTable('location');
+  });
+
+  // Create animal_movement_purpose enum table
+  await knex.schema.createTable('animal_movement_purpose', (table) => {
+    table.increments('id').primary();
+    table.string('key').notNullable();
+  });
+  await knex('animal_movement_purpose').insert(
+    [
+      'GRAZING',
+      'MILKING',
+      'MANURE_MANAGEMENT',
+      'PEST_OR_DISEASE_CONTROL',
+      'BREEDING',
+      'FEEDING',
+      'VETERINARY_CARE',
+      'BIRTHING',
+      'WEANING',
+      'TRANSPORTATION',
+      'BEHAVIORAL_AND_SOCIAL_MANAGEMENT',
+      'EXERCISE',
+      'QUARANTINE',
+      'OTHER',
+    ].map((key) => ({ key })),
+  );
+
+  // Create animal_movement_task table
+  await knex.schema.createTable('animal_movement_task', (table) => {
```

Comment:
Since we aren't storing toLocation ... should we still keep this table for future? its sort of empty right now.

Reply:
We need it to have `{ animal_movement_task: { purposes... } }` in tasks!

Reply:
Haha just for the graph data structure that is interesting!